modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/CopyingServletOutputStream.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ecm.platform.web.common.requestcontroller.filter;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+
+/**
+ * Captures content written to the target stream.
+ *
+ * @since 11.5
+ */
+public class CopyingServletOutputStream extends ServletOutputStream {
+
+    protected final ServletOutputStream output;
+
+    protected final ByteArrayOutputStream capture;
```

Comment:
Good for an initial prototype, but in general we can't buffer in memory.

Reply:
Reworked to rely on apache DeferredFileOutputStream